mailgun/resource_mailgun_domain.go

```diff
@@ -493,3 +491,15 @@ func ImportStatePassthroughDomain(d *schema.ResourceData, meta interface{}) ([]*
 	}
 	return []*schema.ResourceData{d}, nil
 }
+
+func getIps(ctx context.Context,mg *mailgun.MailgunImpl) ([]mailgun.IPAddress, error){
+	start := time.Now()
+	t := time.Now()
+	ipAddress, err := mg.ListDomainIPS(ctx)
+	for (err != nil && t.Sub(start)< 120 * time.Second) {
```

Comment:
What do you think about using the `retry` helper function provided by Terraform: https://github.com/hashicorp/terraform/blob/master/helper/resource/wait.go#L10?

There's an example usage here: https://github.com/terraform-providers/terraform-provider-vault/blob/d8921672d5e07aa78e4a6497478fafaa1b707537/vault/data_source_aws_access_credentials.go#L129-L169

Reply:
I will try.